services/pull/pull.go

```diff
@@ -38,73 +37,70 @@ import (
 var pullWorkingPool = sync.NewExclusivePool()
 
 // NewPullRequest creates new pull request with labels for repository.
-func NewPullRequest(ctx context.Context, repo *repo_model.Repository, pull *issues_model.Issue, labelIDs []int64, uuids []string, pr *issues_model.PullRequest, assigneeIDs []int64) error {
-	if err := TestPatch(pr); err != nil {
-		return err
-	}
-
-	divergence, err := GetDiverging(ctx, pr)
+func NewPullRequest(ctx context.Context, repo *repo_model.Repository, issue *issues_model.Issue, labelIDs []int64, uuids []string, pr *issues_model.PullRequest, assigneeIDs []int64) error {
+	prCtx, cancel, err := createTemporaryRepoForPR(ctx, pr)
```

Comment:
No idea why this change is a must.

The `TestPatch` could be reused by adding a `ctx`

Reply:
> No idea why this change is a must.
> 
> The `TestPatch` could be reused by adding a `ctx`

Because both TestPatch and GetDiverging will invoke `createTemporaryRepoForPR` which should be create only once.